### PR TITLE
Resolves #49: Log the slow query based on how long the query takes to run.

### LINKED ISSUE #49: Slow query logging based on query time
We have slow query logging, which logs the query and plan if the plan contains a table scan. We measure query time for metrics, we should also log slow queries taking time more than a certain threshold.

### PR DESCRIPTION
**Pull request by dongxinEric**

### CODE REVIEW

**Comment by dongxinEric** [comment]
This will resolve #49

File: src/Constants.h [hunk 1]
```diff
@@ -89,6 +89,8 @@ extern const char* MT_TIME_INSERT_LATENCY_US;
 extern const char* MT_HIST_INSERT_SZ;
 extern const char* MT_HIST_TR_PER_REQUEST;
 extern const char* MT_RATE_IDX_REBUILD;
+// TODO: make this part of the knob/options instead of constants?
+extern const uint64_t SLOW_QUERY_THRESHOLD_SECONDS;
```
Comment (apkar) [inline src/Constants.h]: Can we make it knob now? Probably we can also make `slowQueryLogging` a knob.

File: src/ExtMsg.actor.cpp [hunk 2]
```diff
@@ -514,8 +514,15 @@ ACTOR static Future<Void> doRun(Reference<ExtMsgQuery> query, Reference<ExtConne
 		}
 	}
 
-	DocumentLayer::metricReporter->captureTime(DocLayerConstants::MT_TIME_QUERY_LATENCY_US,
-	                                           (timer_int() - startTime) / 1000);
+	uint64_t queryLatencySeconds = (timer_int() - startTime) / 1000;
```
Comment (apkar) [inline src/ExtMsg.actor.cpp]: `time_init()` gives nanoseconds. That's why the metric is in microseconds, `US`, after division. This code fires slow query logging for every request taking more than 10 millies.

File: src/ExtMsg.actor.cpp [hunk 3]
```diff
@@ -514,8 +514,15 @@ ACTOR static Future<Void> doRun(Reference<ExtMsgQuery> query, Reference<ExtConne
...
+	DocumentLayer::metricReporter->captureTime(DocLayerConstants::MT_TIME_QUERY_LATENCY_US, queryLatencySeconds);
+
+	if (slowQueryLogging && queryLatencySeconds >= DocLayerConstants::SLOW_QUERY_THRESHOLD_SECONDS) {
+		TraceEvent("SlowQuery")
+		    .detail("Threshold", DocLayerConstants::SLOW_QUERY_THRESHOLD_SECONDS)
+		    .detail("Duration", queryLatencySeconds)
```
Comment (apkar) [inline src/ExtMsg.actor.cpp]: Can we suffix time unit to the detail type? Something like "DurationSeconds" or "DurationMicroSeconds"?

File: src/ExtMsg.actor.cpp [hunk 4]
```diff
@@ -514,8 +514,15 @@ ACTOR static Future<Void> doRun(Reference<ExtMsgQuery> query, Reference<ExtConne
 		}
 	}
 
-	DocumentLayer::metricReporter->captureTime(DocLayerConstants::MT_TIME_QUERY_LATENCY_US,
-	                                           (timer_int() - startTime) / 1000);
+	uint64_t queryLatencySeconds = (timer_int() - startTime) / 1000;
+	DocumentLayer::metricReporter->captureTime(DocLayerConstants::MT_TIME_QUERY_LATENCY_US, queryLatencySeconds);
+
+	if (slowQueryLogging && queryLatencySeconds >= DocLayerConstants::SLOW_QUERY_THRESHOLD_SECONDS) {
+		TraceEvent("SlowQuery")
```
Comment (apkar) [inline src/ExtMsg.actor.cpp]: This `TraceEvent` could be a `SevWarn`, what do you think?

File: src/Constants.h (hunk 1 above)
Comment (dongxinEric) [reply src/Constants.h]: Sure

File: src/ExtMsg.actor.cpp (hunk 4 above)
Comment (dongxinEric) [reply src/ExtMsg.actor.cpp]: Yep

**Pull request merged**